### PR TITLE
ISLE and mingw32 fixes

### PR DESCRIPTION
**Pull request by madebr**
- My `ISLE.EXE` has a `MXDSAction::setAtom` at `0x00401c40`.
- By implementing the MxOmniCreateParamBase` destructor inline, you don't need to compile mxomnicreateparambase.cpp twice (once for the .EXE and once for the .DLL).
I didn't remove it from the VS project because I don't have a MSVC 4.20 gui set-up.
- Format a few functions + use more macro's in various places.
- I think the `MOUSEMOVE` notification id is `10` instead of `0x10`.
- Apply some changes such that mingw32 can compile the sources (gcc 12.2).
I think being able to build with modern toolchains is useful to avoid creating new bugs.
mingw32 cannot link the `.exe` due to missing symbols: GNU and MSVC has different name mangling. The virtual function table is probably also incompatible.

### CODE REVIEW

File: ISLE/isle.cpp [hunk 1]
```diff
@@ -39,7 +39,8 @@ Isle::Isle()
   rect.m_right = 639;
   rect.m_bottom = 479;
 
-  m_videoParam = MxVideoParam(rect, NULL, 1, MxVideoParamFlags());
+  MxVideoParamFlags video_flags;
+  m_videoParam = MxVideoParam(rect, NULL, 1, video_flags);
```
Comment (itsmattkc) [inline ISLE/isle.cpp:43]: While it's true that GCC won't compile with the original, this produces less accurate code in MSVC420. Accuracy is the priority with this project, modernizations should be kept to other repos/forks.

File: ISLE/main.cpp [hunk 2]
```diff
@@ -49,7 +49,7 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine
   // Throw error if sound unavailable
   if (!soundReady) {
     MessageBoxA(NULL, "\"LEGO\xAE Island\" is not detecting a DirectSound compatible sound card.  Please quit all other applications and try again.",
-      "Lego Island Error",0);
+      "Lego Island Error", MB_OK | MB_APPLMODAL);
```
Comment (itsmattkc) [inline ISLE/main.cpp:52]: Is there a purpose to specifying `MB_APPLMODAL` explicitly?

Comment (madebr) [reply ISLE/main.cpp:52]: No purpose, both `MB_OK` and `MB_APPLMODAL` are defined as `0`.

File: ISLE/isle.cpp (hunk 1 above)
Comment (madebr) [reply ISLE/isle.cpp:43]: Aight, revert it will be.

Comment (madebr) [reply ISLE/isle.cpp]: I've rebased on current master, and force pushed it.
If you prefer reverts to avoid re-reviews, let it know.

Comment (madebr) [reply ISLE/isle.cpp]: About accuracy, both implementations create the flags on the stack, and the variable is not used afterwards.
So the compiler should be able to emit identical assembly.

**Pull request merged**